packages/salesforce-adapter/src/group_changes.ts

```diff
@@ -0,0 +1,38 @@
+/*
+*                      Copyright 2020 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import wu from 'wu'
+import { collections } from '@salto-io/lowerdash'
+import {
+  Change, ChangeGroupIdFunction, getChangeElement, InstanceElement,
+} from '@salto-io/adapter-api'
+import { apiName, isInstanceOfCustomObject } from './transformers/transformer'
+
+export const getInstancesOfCustomObjectChangeGroupIds: ChangeGroupIdFunction = async changes => {
+  const isInstanceOfCustomObjectChange = (
+    change: [collections.set.SetId, Change]
+  ): change is [collections.set.SetId, Change<InstanceElement>] => {
+    const changeElement = getChangeElement(change[1])
+    return isInstanceOfCustomObject(changeElement)
+  }
```

Comment:
I think we have a type called `ChangeEntry` specifically for functions like this
```suggestion
  const isInstanceOfCustomObjectChange = (
    change: ChangeEntry
  ): change is ChangeEntry<InstanceElement> => {
    const changeElement = getChangeElement(change[1])
    return isInstanceOfCustomObject(changeElement)
  }
```